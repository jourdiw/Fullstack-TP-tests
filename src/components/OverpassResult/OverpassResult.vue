<template>
  <textarea v-model='result'></textarea>
</template>

<script>
export default {
  name: 'OverpassResult',
  props: ['result'],
};
</script>

<style scoped>
textarea {
  width: 100%;
  height: 10rem;
}
</style>
