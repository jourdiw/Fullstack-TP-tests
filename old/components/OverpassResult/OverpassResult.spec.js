/* global describe it expect beforeEach beforeAll */
import { mount } from '@vue/test-utils';
import OverpassResult from './OverpassResult.vue';

describe('OverpassResult.spec.js', () => {
  beforeAll(() => {
    console.log('pouic');
  });

  beforeEach(() => {
    console.log('pouet');
  });

  it('is a Vue instance', () => {
    const vm1 = mount(OverpassResult);
    expect(vm1.isVueInstance()).toBeTruthy();
    expect(vm1.is('textarea')).toBe(true);
  });

  it('has the right data to display', () => {
    const wrapper = mount(OverpassResult, {
      propsData: {
        result: 'Hello World',
      },
    });
    expect(wrapper.props().result).toBe('Hello World');
  });

  it('has the expected html structure', () => {
    const vm3 = mount(OverpassResult, {
      propsData: {
        result: 'Hello World 2',
      },
    });
    expect(vm3.element).toMatchSnapshot();
  });
});
