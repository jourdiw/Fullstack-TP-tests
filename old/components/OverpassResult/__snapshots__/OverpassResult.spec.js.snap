// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverpassResult.spec.js has the expected html structure 1`] = `<textarea />`;
